riyaz/disk.py

```diff
@@ -235,23 +235,44 @@ def _load_lesson(
         return db_lesson
 
     def _load_author(
-        self, course_id: int, author: models.Author
+        self, author: models.Author
     ) -> db.Instructor:
         fields = dict(
-            course_id=course_id,
             key=author.key,
             name=author.name,
             about=author.about,
-            photo_path=author.photo,
         )
 
         if db_instructor := db.Instructor.find(key=author.key):
             db_instructor.update(**fields)
         else:
             db_instructor = db.Instructor(**fields)
 
+        db_instructor.save()
+        self._set_instructor_photo(db_instructor, author.photo)
+
         return db_instructor
 
+    def _set_instructor_photo(
+        self,
+        instructor: db.Instructor,
+        on_disk_photo: Optional[Union[FilePath, HttpUrl]],
+    ) -> Optional[db.Asset]:
+        if on_disk_photo is None:
+            instructor.photo_path = None
+            instructor.save()
+            return None
+
+        filename = on_disk_photo.name
```

Comment:
What if if the `on_disk_photo` is an `HttpUrl`?

Reply:
Good catch! I forgot that case after writing the type annotations.
I'll add it now.

Reply:
Removed support for HttpUrl.